c2corg_api/views/route.py

```diff
@@ -282,3 +312,206 @@ def set_title_prefix_for_ids(ids, title):
     """
     DBSession.query(RouteLocale).filter(RouteLocale.id.in_(ids)). \
         update({RouteLocale.title_prefix: title}, synchronize_session=False)
+
+
+def update_all_pt_rating(waypoint_extrapolation=True):
```

Comment:
a bit worried about using the ORM for such a giant query... best way to run this would be in batches ?

Reply:
@eddy-geek 
The first request runs in 3209.054 ms on my laptop
The whole process is done in a few minutes and is supposed to be used only once, afterwards the informations are supposed to be automaticaly updated on route/waypoint creation/updates

With python log level = INFO and postgresql log level to all, the whole process took about 2-3 hours on my laptop and generates huge amount of logs (since that for each route, every linked waypoints are read and the route updated)

Reply:
May be the best is to not expose this function to the REST API and run it with a command line ? So only devs can run it

Reply:
did you run your tests on the full production database?

if it's just a one-time migration then yes we should not leave it in the API, but I guess it's ok to leave it as-is for now and remove the API part after we're done with the initial migration

Reply:
I have run the tests on the docker's data, I don't have access to the production database data

The other way to make the migration is to call the function using a python command